app/controllers/entries_controller.rb

```diff
@@ -55,7 +62,44 @@ def update
 
   def entry_params
     params.require(:entry).permit(
-      :title, :body, :location_id, :country_id, :region_id, :date
+      :title, :body, :latitude, :longitude, :date
     )
   end
+
+  def find_nearest_location(lat1, lon1)
+    locations = Location.all
+    nearest_distance = 1000000
+    nearest_location = locations.first
+    locations.each do |location|
+      distance = distance_miles(lat1, lon1, location.lat, location.lon)
+      if distance < nearest_distance
+        nearest_location = location
+        nearest_distance = distance
+      end
+    end
+    nearest_location
+  end
+
+  def distance_miles( lat1, lon1, lat2, lon2 )
+    rmiles = 3956           # radius of the great circle in miles
+    rad_per_deg = 0.017453293
+    dlon = lon2 - lon1
+    dlat = lat2 - lat1
+
+    dlon_rad = dlon * rad_per_deg
+    dlat_rad = dlat * rad_per_deg
+
+    lat1_rad = lat1 * rad_per_deg
+    lon1_rad = lon1 * rad_per_deg
+
+    lat2_rad = lat2 * rad_per_deg
+    lon2_rad = lon2 * rad_per_deg
+
+    a = (Math.sin(dlat_rad/2))**2 + Math.cos(lat1_rad) *
+         Math.cos(lat2_rad) * (Math.sin(dlon_rad/2))**2
+    c = 2 * Math.atan2( Math.sqrt(a), Math.sqrt(1-a))
+
+    dMi = rmiles * c      # delta between the two points in miles
```

Comment:
Useless assignment to variable - `dMi`.<br>Unnecessary spacing detected.<br>Use snake_case for variable names.